src/social_wallet_api/handler.clj

```diff
@@ -293,15 +300,32 @@ Creates a transaction.
                                                         (:amount query)
                                                         (:to-id query)
                                                         (dissoc query :tags))]
-                           ;; store to db as well with transaction-id
-                           (lib/create-transaction (get-db-blockchain blockchains)
-                                                   (:from-id query)
-                                                   (:amount query)
-                                                   (:to-id query)
-                                                   (-> query 
-                                                       (dissoc :comment :comment-to)
-                                                       (assoc :transaction-id transaction-id)
-                                                       (assoc :currency (:blockchain query))))
+                           (do
+                             ;; Update fee to db when confirmed
+                             ;; The logged-future will return an exception which otherwise would be swallowed till deref
+                             (log/logged-future
+                              (while (> (:number-confirmations blockchain)
```

Comment:
what happens to blockchains that have no need to check confirmations? (like `mongo` or `in-memory`)?
shouldn't there be a flag checked?

Reply:
This is actually checked above. These are actions only for blockchains other than mongo. I will make the code clearer. I have tested and works for both